package.json

```diff
@@ -75,6 +75,7 @@
     "eslint-plugin-import": "^2.27.5",
     "eslint-plugin-react": "^7.32.2",
     "eslint-plugin-react-hooks": "^4.6.0",
+    "expect-type": "^1.2.1",
```

Comment:
I'm using [`expect-type`(used by vitest internally)](https://vitest.dev/guide/features.html#type-testing) here to ensure type safety in our type tests for useSuspenseQuery, useSuspenseQueries, queryOptions.

<img width="729" alt="image" src="https://github.com/user-attachments/assets/b1c30b20-1352-49d1-8e9d-1d49c99ba541" />